Notebook with refactored model for multiple GPUs

expz opened the pull request:
The code is mostly adapted from https://github.com/elbayadm/attn2d.

It changes the dataset and data loading data structures to rely on the `pytorch` standard classes. It also fixes the way most functions were called with an opaque `params` dictionary to use named arguments.

**The code is untested!**

expz committed:
Created notebook with refactored pervasive attention network code more suitable for training on multiple GPUs

ninalopatina commented:
- [x]  I should be added as a reviewer, and you as assignee. Please adhere to this standard for the next PR. 
- [x] Typically I would not approve a PR with un-tested code; please test the code for the next PR.
- [x] ReadME should describe what the project does
- [x] repo should not be named insight. It should have an informative name that tells you what it does.
- [x] venv should also have an informative name, such as the informative name you change your repo to. 
- [x] step 3, gunicorn app:api worked on my local machine, but not my ubuntu machine on mesosphere. Please specify the limitations to what type of machine this can run on. 
- [x] once I do step 3, I lose the terminal window w/the virtualenv. Please include instructions for how to activate again. Now when I try to run step 2's virtualenv --python=/usr/bin/python3 venv, I get the error "The path /usr/bin/python3 (from --python=/usr/bin/python3) does not exist" 
- [x] I have not run the notebook since it is untested. However, I can see that this is not the appropriate use of a notebook. Functions and classes should be defined in .py files and imported into the notebook. Then they can be imported into the notebook and run from there, as a way to experiment during dev or to share results with someone viewing the notebook. Ideally, no notebooks would be used, and this would all be in python scripts that can be run from the command line.

expz commented:
@ninalopatina  I looked into limitations for where gunicorn can run, and I cannot find any.
* I checked the documentation for install, running and deploying, and it does not mention any special requirements.
* I do not have an experience with it failing to work when I used it at my previous job.
So I cannot guess what is not working without more information about the error.

The only two general things I can think of that maybe you already know:
* If you move a virtual environment, it breaks. So if you followed the setup directions locally and then pushed the directory to mesosphere, it would break it. I added a comment about this in the README.
* If you are running another app on port 8000, then gunicorn will not be able to host the app on that port. (This is not specific to gunicorn--it is how ports work.) I changed the port to be 8887, which I have never seen another project use, to reduce the probability of a port collision.
